### ISSUE TITLE
Agent works wrong in parallel mode

### ISSUE THREAD

**Issue by Invictum** [reportportal/agent-js-mocha#49]
Seems like agent works not as expected in parallel mode.

1. Suites structure is wrong in RP

Test files:
```
// one.test.js
describe('block-one', function () {
    it('one-one', function () {
        
    })
    it('one-two', function () {

    })
})

// two.test.js
const rp = require('@reportportal/agent-js-mocha/lib/publicReportingAPI');

describe('block-two', function () {
    it('two-one', function () {
        rp.log('INFO', `two-one, pid: ${process.pid}`)
    })
    it('two-two', function () {
        rp.info(`two-two, pid: ${process.pid}`)
    });
})
```
With `parallel` flag reporter produces launch with 4 tests and ignores suites defined in describe block
```
Launch
.
+-- one-one
+-- one-two
+-- two-one
+-- two-two
```
Without `parallel` flag it works as expected and builds identical structure in RP
```
Launch
.
+-- block-one
        .
        +-- one-one
        +-- one-two
+-- block-two
        .
        +-- two-one
        +-- two-two

```

2. Calls of methods of `PublicReportingAPI` doesn't produce any output in RP report (log entries, test attributes changing)

**Comment by AmsterGet** [comment]
Hi @Invictum !
Currently, `agent-js-mocha` does not support reporting for parallel mode.
We are already planning this feature for the next minor version of the agent `5.1`.
Keep up to date with our [slack](https://reportportal-slack-auto.herokuapp.com/) (`#agent_js` channel).

**Comment by dnlstpnv** [comment]
@AmsterGet Hi! Any updates for this?

**Comment by AmsterGet** [comment]
Hello @dnlstpnv !
We are currently working on a new mechanism for combining several launches into one, which will be used in all JS agents.
ETA at the moment is the end of the Q2-Q3 of 2023.
Since this is a common request for all integrations, I plan to create an aggregated GitHub issue to track progress.
I will keep you updated as soon as it is ready.

**Comment by dnlstpnv** [comment]
Hello @AmsterGet , can we expect good news in the near future?)

**Comment by AmsterGet** [comment]
Hello @dnlstpnv !
Sorry for the news that is not as good as you expected, but our plans are constantly changing, we need more time to prepare this huge update to the JS integration ecosystem.
We agreed that a dedicated person from the team will work on the JS integrations, so I hope it will make the development and support process more predictable.
At the moment, the deadlines have been postponed to the end of the 4th quarter.
I hope for your understanding and thank you for your interest.

In the meantime, several workarounds are available. Please check the issue from [agent-js-playwright](https://github.com/reportportal/agent-js-playwright/issues/86#issuecomment-1490687597).
